fairlearn/reductions/_exponentiated_gradient/_lagrangian.py

```diff
@@ -149,7 +150,10 @@ def _call_oracle(self, lambda_vec):
                                          constant=redY_unique[0])
             self.n_oracle_calls_dummy_returned += 1
         else:
-            classifier = pickle.loads(self.pickled_estimator)
+            # use sklearn.base.clone to clone the estimator.
+            # Use safe=False to make a deep copy of the estimator if not an
+            # sklearn-based estimator.
```

Comment:
Relieved to see that this will work with non-sklearn estimators. Can you let us know what this offers over `copy.deepcopy` (which would have been my first guess for an improvement over pickling)?

Reply:
It is actually just the same as `copy.deepcopy` for non-sklearn estimators. But for sklearn estimators, it would just be more efficient as it would not really do a complete `copy.deepcopy` for them. It would just instantiate a new estimator using the `get_params()` parameters internally. The [implementation](https://github.com/scikit-learn/scikit-learn/blob/fd237278e/sklearn/base.py#L48) of `sklearn.base.clone` will make it more clear.

Reply:
Could you put that explanation in the description? I'm all for getting rid of the pickling.

Reply:
Also please add a note to CHANGES.md to indicate that we're using this instead of pickle.